src/main/java/de/rwth/dbis/neologism/recommender/Query.java

```diff
@@ -45,8 +50,8 @@ public Query(String query, Model context) {
     }
 
     public Query(String query, Model context, int limit) {
-        this.context = Preconditions.checkNotNull(context);
-
+        // this.context = Preconditions.checkNotNull(context);
```

Comment:
The context not being null is a precondition for many of the existing processors.

Reply:
Good point. @enisze is it possible to overload that method for your needs? That is, keep the existing method intact (with the Null check active), and add a second one with the same name for your needs, like `public Query(String query, int limit)`, which you call from your "new" lines of code.

Reply:
I just saw you did something like this. Still, please alternate the method calls or so in order to ensure that the Null check is in place if someone calls this `Query(query,context,limit)` constructor.

Reply:
This is about the contracts of the class. Just creating a new constructor does not solve this issue.

Reply:
this was a very old state - i returned the changes :) I am not using Query

Reply:
Thank you for clarifying.